src/components/Shop/Modal/PurchaseModal.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext, FC } from "react";
+import styled from "styled-components";
+import { StateContext } from "state/state";
+import { calculateCartTotal } from "types/inventory";
+import { OffWhite } from "utility/style";
+import Button from "../../../atoms/Button";
+import purchaseItems from "api/order";
+import { setModalState } from "state/actions";
+import { modalTypes } from "types/modal";
+
+const styledButton = {
+  gridColumn: 2,
+  gridRow: 3,
+  width: "80%",
+  height: "85%",
+  outline: "none",
+  marginLeft: "auto",
+  marginRight: "auto",
+  boxShadow: "2px 2px 7px #888888",
+};
+
+const styledText = {
+  gridColumn: 2,
+  gridRow: 1,
+};
+
+const PurchaseModal: FC = () => {
+  const { state, dispatch } = useContext(StateContext);
+  const { cart, inventory, user } = state;
```

Comment:
I learned recently that you can destruct like
```const { state: { cart, inventory, user } } = useContext(StateContext);```
which means you probably could do
```
const {
  state: { cart, inventory, user },
  dispatch,
 } = useContext(StateContext);
```

Obviously this is preferrential, but I thought it was cool!

Reply:
Its a really cool feature, but I think is looks a little messy. Perhaps we can look into it in the future, if we do some refactoring later on!